Frameworks/AudioToolbox/SystemSound.mm

```diff
@@ -16,62 +17,337 @@
+
+            TraceInfo(TAG, L"Loading media at file URL: %hs\n", [appxurl.absoluteString UTF8String]);
+
+            _mediaUri = [WFUri makeUri:[NSString stringWithFormat:@"ms-appx://%@", appxurl.path]];
+        } else {
+            _mediaUri = [WFUri makeUri:url.absoluteString];
+
+            TraceInfo(TAG, L"Loading media at URI: %hs\n", [_mediaUri.absoluteUri UTF8String]);
+        }
+    }
+
+    return self;
+}
+
+
+- (void)runCallbackFunction {
+
+    if (_callbackFunction != nullptr) {
+        dispatch_async(dispatch_get_main_queue(), ^{ _callbackFunction(_systemSoundID, _clientData); });
+    }
+}
+
+
+- (void)setCallbackFunction:(AudioServicesSystemSoundCompletionProc)callbackFunction withClientData:(void*)clientData {
+    self.callbackRegistered = true;
+    _callbackFunction = callbackFunction;
+    _clientData = clientData;
+}
+
+
+- (void)removeCallbackFunction {
+    self.callbackRegistered = false;
+    _callbackFunction = nil;
+    _clientData = nil;
+}
+@end
+
+
+// Runs the callback function, if any, of the latest playing audio file.
+void _handleMediaEndedEvent() {
+
+    if (_latestSystemSoundID > c_systemSoundIDStart) {
+        SystemSound* systemSoundObject = _systemSoundDictionary[[NSNumber numberWithUnsignedInt:_latestSystemSoundID]];
+
+        std::lock_guard<std::mutex> lock(_mutex);
+        _latestSystemSoundID = 0;
+
+        if (systemSoundObject) {
+            [systemSoundObject runCallbackFunction];
+        }
+    }
+}
+
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 OSStatus AudioServicesCreateSystemSoundID(CFURLRef inFileURL, SystemSoundID* outSystemSoundID) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    static SystemSoundID lastSystemSoundID = c_systemSoundIDStart;
+
+    if (inFileURL) {         
+        SystemSound* systemSoundObject = [[SystemSound alloc] initWithContentsOfURL:inFileURL forSystemSoundID:++lastSystemSoundID];
+        *outSystemSoundID = lastSystemSoundID;
+        [_systemSoundDictionary setObject:systemSoundObject forKey:[NSNumber numberWithUnsignedInt:*outSystemSoundID]];
+        return kAudioServicesNoError;
+    }
+    
+    return kAudioServicesSystemSoundUnspecifiedError;
 }
 
+
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 OSStatus AudioServicesDisposeSystemSoundID(SystemSoundID inSystemSoundID) {
-    UNIMPLEMENTED();
-    return StubReturn();
+
+    if (inSystemSoundID > c_systemSoundIDStart) {
+
+        // This insures _numberOfCallbacksRegistered variable is decremented.
+        AudioServicesRemoveSystemSoundCompletion(inSystemSoundID);
+
+        [_systemSoundDictionary removeObjectForKey:[NSNumber numberWithUnsignedInt:inSystemSoundID]];
+        return kAudioServicesNoError;
+    }
+
+    return kAudioServicesSystemSoundUnspecifiedError;
+}
+
+
+// Calls into C++ to access Windows.Phone.Devices.Notification namespace to vibrate devices with vibration capability.
```

Comment:
As discussed the projections for Windows Phone are currently not available, Issue #480. Hence we use WRL